de.tudarmstadt.ukp.wikipedia.api/src/test/java/de/tudarmstadt/ukp/wikipedia/api/CategoryTest.java

```diff
@@ -45,150 +48,353 @@ public static void setupWikipedia() {
+        } catch (WikiPageNotFoundException pnfe) {
+            // this is expected behavior here, provoked by the test
+        }
+    }
+
+    @Test
+    public void testCreateCategoryByName() {
+        try {
+            Category p = new Category(wiki, A_FAMOUS_CATEGORY);
+            assertNotNull(p);
+            assertEquals(A_FAMOUS_CATEGORY, p.getTitle().getPlainTitle());
+            assertEquals(A_FAMOUS_PAGE_ID, p.getPageId());
+        } catch (WikiApiException e) {
+            fail("A WikiApiException occurred creating a page: " + e.getLocalizedMessage());
+        }
+    }
+
+    @Test
+    public void testCreateCategoryByNameRandom() {
+        try {
+            new Category(wiki, UUID.randomUUID().toString());
+        } catch (WikiApiException e) {
+            // this is expected, as a random page title should not be found
+        }
+    }
+
+    @Test
+    public void testCreateCategoryByNameEmpty() {
+        try {
+            new Category(wiki, "");
+        } catch (WikiPageNotFoundException pnfe) {
+            // this is expected behavior here, provoked by the test
+        } catch (WikiApiException e) {
+            fail("A WikiApiException occurred creating a page: " + e.getLocalizedMessage());
+        }
+    }
+
+    @Test
+    public void testCreateCategoryByNameNull() {
+        try {
+            new Category(wiki, null);
+        } catch (WikiPageNotFoundException pnfe) {
+            // this is expected behavior here, provoked by the test
+        } catch (WikiApiException e) {
+            fail("A WikiApiException occurred creating a page: " + e.getLocalizedMessage());
+        }
+    }
+
+    @Test
+    public void testGetCategoryInfo() {
+        try {
+            Category p = new Category(wiki, "UKP");
+            assertNotNull(p);
+            String categoryInfo = p.getCategoryInfo();
+            assertNotNull(categoryInfo);
+            assertTrue(categoryInfo.length() > 0);
+        } catch (WikiApiException e) {
+            fail("A WikiApiException occurred while getting the page info: " + e.getLocalizedMessage());
+        }
+    }
+
+    @Test
+    public void testCategorySiblings(){
+        Category cat;
+        try {
+            cat = wiki.getCategory(A_FAMOUS_CATEGORY);
+            assertNotNull(cat);
+            Set<Integer> expectedPageIds = new HashSet<Integer>();
+            Set<Integer> isIds = new HashSet<Integer>();
+            //test the children
+            expectedPageIds.clear();
+            expectedPageIds.add(7);
+            expectedPageIds.add(8);
+            expectedPageIds.add(9);
+            Set<Category> siblings = cat.getSiblings();
+            assertNotNull(siblings);
+            assertTrue(siblings.size() > 0);
+            for(Category sibling : siblings) {
+                isIds.add(sibling.getPageId());
+            }
+            assertEquals("siblings",expectedPageIds,isIds);
+        } catch (WikiApiException e) {
```

Comment:
General comment: there is no need to catch exceptions and handle them by failing explicitly. Just make the unit test method declare `throws Exception`.

Reply:
see other comment on Exceptions/Tests.